internal/util/apiclient/error_handler.go

```diff
@@ -33,5 +39,17 @@ func HandleErrorResponse(res *http.Response, requestErr error) error {
 		return errors.New(string(body))
 	}
 
+	checkVersionsMismatch(res, errResponse)
+
 	return errors.New(errResponse.Error)
 }
+
+func checkVersionsMismatch(res *http.Response, err ApiErrorResponse) {
+	// Ignore check if error comes from health-check response
+	if !strings.Contains(err.Error, "failed to check server health at:") {
```

Comment:
I suggest we use the `IsERROR` "pattern". (e.g. `IsHealthCheckFailed(err)` where `IsHealthCheckFailed` preforms this check + the `GetApiClient` function can then return `ErrHealthCheckFailed(...)`). This is more maintainable because a change in the health check message won't require a change here.